isis/src/socet/apps/socetframesettings/socetframesettings.cpp

```diff
@@ -471,110 +683,284 @@ void getCamPosOPK(Spice &spice, QString spacecraftName, double et, Camera *cam,
+
+
+       +Yss  ^
+             |
+             |
+             |
+             |+Zss (out of screen)
+           --O-----------------------------------> +SAMPLEss
+             |                                     +Xss
+             |
+             |      X--------------------------------------------------+
+             |      |(1,1)ss                                           |
+             |      |                                                  |
+    +LINEss  V      |       _                                          |
+                    |      | \                                         |
+                    |      |    *                                      |
+                    |      |           |     ___                       |
+                    |      |    |    --+--  /   \                      |
+                    |    --+--  |      |    \___                       |
+                    |      |    |      |        \                      |
+                    |      |    |_/    |_/  \___/                      |
+                    |                                                  |
+                    |                                                  |
+                    |       _    ___    _    ___                       |
+                    |      | \  /   \  | \  /   \                      |
+                    |      |     ___/  |     ___/                      |
+                    |      |    /      |    /                          |
+                    |      |    \___/  |    \___/                      |
+                    |                                                  |
+ +NAXIS2fits ^      |      *           *                               |
+ +LINEisis   |      |                                                  |
+ +Xfits      |      |                                                  |
+ +Xisis      |      |(1,1)isis == (0,0)fits                            |
+             |      X--------------------------------------------------+
+             |
+             |
+             |+Zfits = +Zisis (into screen)
+          ---X---------------------------------> +NAXIS1fits
+             |                                   +SAMPLEisis
+             |                                   +Yfits      
+             |                                   +Yisis      
+
+ +Xss == +Yisis == +Yfits
+ +Yss == +Xisis == +Xfits
+ +Zss == -Zisis == -Zfits
+
+   ********************************************************************/
+
+  if (spacecraftName == "OSIRIS-REX") {
+    /* MapCam and PolyCam ISIS-to-SS Matrix swaps X and Y, inverts Z */
+    isisFocalPlane2SocetPlate[1][0] =  1.0;  // +Xisis => +Yss
+    isisFocalPlane2SocetPlate[0][1] =  1.0;  // +Yisis => +Xss
+    isisFocalPlane2SocetPlate[2][2] = -1.0;  // +Zisis => -Zss
+  }
+
+  // Confirm that matrix is now a rotation matrix
+  if (!isrot_c(isisFocalPlane2SocetPlate, 1e-10, 1e-10)) {
```

Comment:
Checking if the rotation matrix is valid as a proxy for checking if the instrument is supported seems obtuse.  I'd much rather see an if/else or switch setup used here with the default/else case being this exception.  Then, a separate check can be done to verify that the rotation matrix is valid.